src/components/__tests__/ColorPicker.spec.ts

```diff
@@ -0,0 +1,215 @@
+
+            colorPickerComponent.setProps({ type: "block", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".block-picker").length).toBe(1);
+        });
+
+        it("github", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "github", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".github-picker").length).toBe(1);
+        });
+
+        it("twitter", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "twitter", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".twitter-picker").length).toBe(1);
+        });
+
+        it("circle", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "circle", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".circle-picker").length).toBe(1);
+        });
+
+        it("hue", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "hue", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".hue-picker").length).toBe(1);
+        });
+
+        it("alpha", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "alpha", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".alpha-picker").length).toBe(1);
+        });
+
+        it("slider", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "slider", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".slider-picker").length).toBe(1);
+        });
+
+        it("compact", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "compact", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".compact-picker").length).toBe(1);
+        });
+
+        it("material", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "material", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".material-picker").length).toBe(1);
+        });
+
+        it("swatches", () => {
+            const colorPickerComponent = fullRenderColorPicker(colorPickerProps);
+            colorPickerComponent.setProps({ type: "swatches", displayColorPicker: true });
+
+            expect(colorPickerComponent.find(".swatches-picker").length).toBe(1);
+        });
+    });
+
+    describe("with a mode as", () => {
```

Comment:
should we have 3 separate test, for popover, inline and button?

Reply:
This component takes button and input as children some the rendering here is the same for the two